src/averageAges.js

```diff
@@ -56,6 +74,20 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const mothers = people.filter(person => person.sex === 'f'
+  && people.some(women => women.mother === person.name));
+
+  const children = people.filter(person => onlyWithSon
+    ? mothers.some(mother => mother.name === person.mother)
+  && person.sex === 'm'
+    : mothers.some(mother => mother.name === person.mother));
+
+  const ageDifference = children.reduce((totalAgeDiff, child) =>
+    totalAgeDiff + (child.born - (mothers.find(mother => mother.name
+  === child.mother)).born), 0
+  ) / children.length;
```

Comment:
it's better to create an array of ageDifference and then find it's average, now this code looks messy